src/panels/lovelace/cards/hui-alarm-panel-card.js

```diff
@@ -0,0 +1,232 @@
+          </div>
+          <div>
+            <paper-button noink raised value='2' on-click='_handlePadClick'>2</paper-button>
+            <paper-button noink raised value='5' on-click='_handlePadClick'>5</paper-button>
+            <paper-button noink raised value='8' on-click='_handlePadClick'>8</paper-button>
+            <paper-button noink raised value='0' on-click='_handlePadClick'>0</paper-button>
+          </div>
+          <div>
+            <paper-button noink raised value='3' on-click='_handlePadClick'>3</paper-button>
+            <paper-button noink raised value='6' on-click='_handlePadClick'>6</paper-button>
+            <paper-button noink raised value='9' on-click='_handlePadClick'>9</paper-button>
+            <paper-button noink raised value='clear' on-click='_handlePadClick'>CLEAR</paper-button>
+          </div>
+      </template>
+      <template is='dom-if' if='[[!_stateObj]]'>
+        <div>Entity not available: [[_config.entity]]</div>
+      </template>
+    </ha-card>
+    `;
+  }
+
+  static get properties() {
+    return {
+      hass: {
+        type: Object
+      },
+      _config: Object,
+      _stateObj: {
+        type: Object,
+        computed: '_computeStateObj(hass.states, _config.entity)',
+      },
+      _value: {
+        type: String,
+        value: ''
+      },
+    };
+  }
+
+  getCardSize() {
+    return 4;
+  }
+
+  setConfig(config) {
+    if (!config || !config.entity || config.entity.split('.')[0] !== 'alarm_control_panel') throw new Error('Invalid card configuration');
+    this._config = Object.assign({ states: ['arm_away', 'arm_home'] }, config);
+    this._icons = {
+      armed_away: 'hass:security-lock',
+      armed_custom_bypass: 'hass:security',
+      armed_home: 'hass:security-home',
+      armed_night: 'hass:security-home',
+      disarmed: 'hass:verified',
+      pending: 'hass:shield-outline',
+      triggered: 'hass:bell-ring',
+    };
+  }
+
+  _computeStateObj(states, entityId) {
+    return states && entityId in states ? states[entityId] : null;
+  }
+
+  _computeHeader(stateObj) {
+    if (!stateObj) return '';
+    return this._config.title ? this._config.title : this._label(stateObj.state);
+  }
+
+  _computeStateText(stateObj) {
+    return this._config.title ? this._label(stateObj.state) : '';
+  }
+
+  _computeIcon(stateObj) {
+    return this._icons[stateObj.state] || 'hass:shield-outline';
+  }
+
+  _label(state) {
+    return state.split('_').join(' ').replace(/^\w/, c => c.toUpperCase());
+  }
+
+  _stateIconLabel(state) {
+    const stateLabel = state.split('_').pop();
+    return stateLabel === 'disarmed' || stateLabel === 'triggered' ? '' : stateLabel;
+  }
+
+  _showActionToggle(state) {
+    return state === 'disarmed';
+  }
+
+  _computeClassName(stateObj) {
+    if (!stateObj) return 'not-found';
+    return '';
+  }
+
+  _handlePadClick(e) {
+    this._value = e.target.getAttribute('value') === 'clear' ? '' : this._value + e.target.getAttribute('value');
```

Comment:
I would probably shorten this line by setting `const val = e.target.getAttribute('value')` first